lib/board.rb

```diff
@@ -0,0 +1,71 @@
+class Board
+
+	attr_accessor :cells, :token, :player
+
+	def initialize
+		reset!
+	end
+
+	def reset!
+		#How the board looks at the start of a game
+		@cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+	end
+
+	def display
+		puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+		puts "-----------"
+		puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+		puts "-----------"
+		puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+	end
+
+	def position(position)
+		#gets user input through argument
+		#looks up the value
+		cells[position.to_i-1]
+	end
+
+	def turn_count
+		#keeps track of the number of moves that have been played
+		counter = 0
+		cells.each do |count|
+			if count == "X"
+				counter +=1
+			elsif count == "O"
+				counter +=1
+			end
+		end
+		counter
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count {|cell| cell == "X" || cell == "O"}
end
```